api/v1alpha1/releaselink_webhook_unit_test.go

```diff
@@ -0,0 +1,155 @@
+//
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func TestReleaseLinkCreateValidatingWebhook(t *testing.T) {
+
+	tests := []struct {
+		name  string
+		newRL ReleaseLink
+		err   string
+	}{
+		{
+			name: "releaselink namespace cannot be the same as its target",
+			err:  "the target namespace1 and the namespace namespace1 are the same; they must differ",
+			newRL: ReleaseLink{
+				ObjectMeta: v1.ObjectMeta{
+					Namespace: "namespace1",
+				},
+				Spec: ReleaseLinkSpec{
+					DisplayName: "releaselink1",
+					Application: "application1",
+					Target:      "namespace1",
+				},
+			},
+		},
+		{
+			name: "valid releaselink",
+			newRL: ReleaseLink{
+				ObjectMeta: v1.ObjectMeta{
+					Namespace: "namespace1",
+				},
+				Spec: ReleaseLinkSpec{
+					DisplayName: "releaselink1",
+					Application: "application1",
+					Target:      "target1",
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := test.newRL.ValidateCreate()
+
+			if test.err == "" {
+				assert.Nil(t, err)
+			} else {
+				assert.Contains(t, err.Error(), test.err)
+			}
+		})
+	}
+}
+
+func TestReleaseLinkUpdateValidatingWebhook(t *testing.T) {
+
+	originalReleaseLink := ReleaseLink{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "namespace1",
+		},
+		Spec: ReleaseLinkSpec{
+			DisplayName: "releaselink1",
+			Application: "application1",
+			Target:      "target1",
+		},
+	}
+
+	tests := []struct {
+		name     string
+		updateRL ReleaseLink
```

Comment:
I think that the following is good enough. The context (action) is already clear in the function.
```suggestion
		releaseLink ReleaseLink
```